include/wx/graphics.h

```diff
@@ -449,6 +431,95 @@ class WXDLLIMPEXP_CORE wxGraphicsGradientStops
     wxVector<wxGraphicsGradientStop> m_stops;
 };
 
+// ----------------------------------------------------------------------------
+// wxGraphicsPenInfo describes a wxGraphicsPen
+// ----------------------------------------------------------------------------
+
+class wxGraphicsPenInfo : public wxPenInfoBase<wxGraphicsPenInfo>
+{
+public:
+    explicit wxGraphicsPenInfo(const wxColour& colour = wxColour(),
+                               wxDouble width = 1.0,
+                               wxPenStyle style = wxPENSTYLE_SOLID)
+        : wxPenInfoBase<wxGraphicsPenInfo>(colour, style)
+    {
+        m_width = width;
+        m_gradientType = wxGRADIENT_NONE;
+    }
+
+    // Setters
+
+    wxGraphicsPenInfo& Width(wxDouble width)
+        { m_width = width; return *this; }
+
+    wxGraphicsPenInfo& 
+    LinearGradient(wxDouble x1, wxDouble y1, wxDouble x2, wxDouble y2,
+                   const wxColour& c1, const wxColour& c2)
+        { 
+            m_gradientType = wxGRADIENT_LINEAR;
+            m_x1 = x1; m_y1 = y1; m_x2 = x2; m_y2 = y2;
+            m_stops.SetStartColour(c1);
+            m_stops.SetEndColour(c2);
+            return *this; 
+        }                                      
+
+    wxGraphicsPenInfo& 
+    LinearGradient(wxDouble x1, wxDouble y1, wxDouble x2, wxDouble y2,
+                   const wxGraphicsGradientStops& stops)
+        { 
+            m_gradientType = wxGRADIENT_LINEAR;
+            m_x1 = x1; m_y1 = y1; m_x2 = x2; m_y2 = y2;
+            m_stops = stops;
+            return *this; 
+        }
+
+    wxGraphicsPenInfo& 
+    RadialGradient(wxDouble xo, wxDouble yo, wxDouble xc, wxDouble yc, wxDouble radius, 
+                   const wxColour& oColor, const wxColour& cColor)
+        { 
+            m_gradientType = wxGRADIENT_RADIAL;
+            m_x1 = xo; m_y1 = yo; m_x2 = xc; m_y2 = yc;
+            m_radius = radius;
+            m_stops.SetStartColour(oColor);
+            m_stops.SetEndColour(cColor);
+            return *this; 
+        }                                      
+
+    wxGraphicsPenInfo& 
+    RadialGradient(wxDouble xo, wxDouble yo, wxDouble xc, wxDouble yc, 
+                   wxDouble radius, const wxGraphicsGradientStops& stops)
+        { 
+            m_gradientType = wxGRADIENT_RADIAL;
+            m_x1 = xo; m_y1 = yo; m_x2 = xc; m_y2 = yc;
+            m_radius = radius;
+            m_stops = stops;
+            return *this; 
+        }
+
+    // Accessors
+
+    wxDouble GetWidth() const { return m_width; }
+    wxGradientType GetGradientType() const { return m_gradientType; }
+    wxDouble GetX1() const { return m_x1; }
+    wxDouble GetY1() const { return m_y1; }
+    wxDouble GetX2() const { return m_x2; }
+    wxDouble GetY2() const { return m_y2; }
+    wxDouble GetXO() const { return m_x1; }
```

Comment:
I'm not really happy about using `xo` and `yc` for parameter names, but method names should really have more intelligible names IMO.

Reply:
I'm certainly fine with changing the parameter names, they are not part of the API in C++, so doing this can't break anything. Thanks for cleaning this up!

FWIW I think you should be able to handle the old names as synonyms for the new ones in Python, but you certainly know more about this than I do anyhow.